task.go

```diff
@@ -294,14 +303,32 @@ func (t *task) Exec(ctx context.Context, id string, spec *specs.Process, ioCreat
 	}, nil
 }
 
-func (t *task) Pids(ctx context.Context) ([]uint32, error) {
+func (t *task) Pids(ctx context.Context) ([]ProcessInfo, error) {
 	response, err := t.client.TaskService().ListPids(ctx, &tasks.ListPidsRequest{
 		ContainerID: t.id,
 	})
 	if err != nil {
 		return nil, errdefs.FromGRPC(err)
 	}
-	return response.Pids, nil
+	var processList []ProcessInfo
+	for _, p := range response.Processes {
+		processInfo := ProcessInfo{
+			Pid: p.Pid,
+		}
+		if p.Info != nil {
+			v, err := typeurl.UnmarshalAny(p.Info)
+			if err != nil {
+				return nil, errdefs.FromGRPC(err)
+			}
+			t, ok := v.(interface{})
+			if !ok {
+				return nil, errors.New("unsupported ProcessInfo.Info type")
```

Comment:
We should make sure of the unsupported/notimplemented error type here `errors.Wrap(errdefs.ErrNotImplemented, "unsupported ProcessInfo.Info type")`